File: Day15/valid_word.cc
class Solution {
// Time: O(N) | Space: O(1)
public:
    bool isValid(string word) {
        if (word.length() < 3) return false;

        bool foundVowel = false, foundConsonant = false;
        for (char ch : word) {
            if (isalpha(ch)) {
                char lower = tolower(ch);
                if (lower == 'a' || lower == 'e' || lower == 'i' || lower == 'o' || lower == 'u')
                    foundVowel = true;
                else
                    foundConsonant = true;
            } else if (!isdigit(ch)) {
                return false;
            }
        }
        return foundVowel && foundConsonant;
    }
};


class Solution {
// Time: O(N) | Space: O(1)
public:
    bool isValid(string word) {
        if (word.length() < 3) return false;

        vector<char> digits = {'0','1','2','3','4','5','6','7','8','9'};
        vector<char> vowels = {'A','a','E','e','I','i','O','o','U','u'};
        vector<char> symbols = {'@', '#', '$'};

        bool foundVowel = false, foundConsonant = false;

        for (char ch : word) {
            if (find(symbols.begin(), symbols.end(), ch) != symbols.end()) {
                return false;
            }
            if (find(vowels.begin(), vowels.end(), ch) != vowels.end()) {
                foundVowel = true;
            } else if (find(digits.begin(), digits.end(), ch) == digits.end()) {
                foundConsonant = true;
            }
        }

        return foundVowel && foundConsonant;
    }
};

File: Day15/valid_word.py
class Solution:
    # Time: O(N) | Space: O(1)
    def isValid(self, word: str) -> bool:
        if len(word) < 3:
            return False
        foundVowel= foundConsonant = False
        digits = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9']
        vowels = ['A', 'a', 'e', 'E', 'i', 'I', 'O', 'o', 'U', 'u']
        symbols =['@', '#', '$']
        
        for ch in word:
            if ch in symbols:
                return False
            if ch in vowels:
                foundVowel = True
            elif ch not in digits: 
                foundConsonant = True
        
        return  foundVowel and foundConsonant


class Solution:
    # Time: O(N) | Space: O(1)
    def isValid(self, word: str) -> bool:
        if len(word) < 3:
            return False
        foundVowel= foundConsonant = False
        for ch in word:
            if ch.isalpha():
                if ch.lower() in 'aeiou':
                    foundVowel = True
                else:
                    foundConsonant = True
            elif not ch.isdigit():
                return False
        return foundVowel and foundConsonant


        